packages/js/product-editor/src/components/variations-table/variation-actions-menus/variation-actions-menu-item.tsx

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import { Slot, Fill } from '@wordpress/components';
+import { createElement, Fragment } from '@wordpress/element';
+import {
+	createOrderedChildren,
+	sortFillsByOrder,
+} from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import { MenuItemProps } from './types';
+import { QUICK_UPDATE, SINGLE_UPDATE } from './constants';
+
+const getGroupName = ( group?: string, type?: string ) => {
+	const name = 'woocommerce-actions-menu-slot';
+	const nameSuffix = type ? `_${ type }` : '';
+	return group ? `${ name }_${ group }${ nameSuffix }` : name;
+};
+
+export const VariationActionsMenuItem: React.FC< MenuItemProps > & {
+	Slot: React.FC< Slot.Props & { group: string; type: string } >;
+} = ( { children, order = 1, group = '_main', type } ) => {
+	if ( type ) {
+		return (
+			<Fill name={ getGroupName( group, type ) }>
+				{ ( fillProps: Fill.Props ) =>
+					createOrderedChildren( children, order, fillProps )
+				}
+			</Fill>
+		);
+	}
+
+	return (
+		<>
+			{ [ QUICK_UPDATE, SINGLE_UPDATE ].map( ( actionType ) => (
+				<Fill
+					key={ actionType }
+					name={ getGroupName( group, actionType ) }
+				>
+					{ ( fillProps: Fill.Props ) =>
+						createOrderedChildren( children, order, fillProps )
+					}
+				</Fill>
+			) ) }
+		</>
+	);
+};
+
+VariationActionsMenuItem.Slot = ( { fillProps, group = '_main', type } ) => {
+	return (
+		<Slot name={ getGroupName( group, type ) } fillProps={ fillProps }>
+			{ ( fills ) => {
+				if ( ! sortFillsByOrder ) {
+					return null;
+				}
+
+				return sortFillsByOrder( fills );
+			} }
+		</Slot>
+	);
+};
+
+export const VariationSingleUpdateMenuItem = ( {
```

Comment:
I left the components `VariationSingleUpdateMenuItem` and `VariationQuickUpdateMenuItem` here because I considered them as one of the same kind as `VariationActionsMenuItem` (an extensibility component), but I'm totally ok with moving them to other files.
The idea underneath was to have multiple ways to do the same. The component `VariationActionsMenuItem` also provides the possibility of adding items under both action menus.